x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java

```diff
@@ -199,7 +217,17 @@ private <Request, Result> void modifyMapping(
         } else {
             try {
                 logger.trace("Modifying role mapping [{}] for [{}]", name, request.getClass().getSimpleName());
-                inner.accept(request, ActionListener.wrap(r -> refreshRealms(listener, r), listener::onFailure));
+                inner.accept(request, ActionListener.wrap(r -> {
+                    if (lastLoadCacheEnabled) {
+                        logger.trace(
+                            "Clearing last role mapping load since a role mapping [{}] was modified for [{}]",
+                            name,
+                            request.getClass().getSimpleName()
+                        );
+                        lastLoadRef.set(null);
```

Comment:
We could leave this out -- it feels trappy though to not clear the entry in case of a modified role mapping since we know that the list is stale in this case. Let me know if you agree -- I'll add a test case.

Reply:
It seems weird to do this only on the node that modified the mapping.

I don't have a strong view, but there are a few options:
1. Clear the cache (because we know it's out of date)
2. Don't clear the cache (because no other node will, and something is better than nothing)
3. Reload the mappings from the index (which may be unnecessary work) 
4. Be smarter & update the cache in place (which seems like more complexity than is justified).

None of them are great options, but since this just meant to be a backup for broken indices, any of them are probably fine.

Reply:
True, it is weird to clear on just one node -- I propose that we don't clear, in this PR, and possibly add a distributed cache clearing call in a follow-up. For the specific target of this fix, we know there are very few role mapping modification calls (and no deletions) so not clearing feels acceptable and is the simplest thing we can do; if we consistently clear the cache on all nodes, we can consider exposing this to customers.